File: Agile Development Practices/Code/DotNet/videoworld/VideoWorld/Models/RegularPrice.cs
namespace VideoWorld.Models
{
    public class RegularPrice : IPrice
    {
        public decimal GetCharge(int periodInDays)
        {
            decimal result = 2.00m;
            if (periodInDays > 2)
                result += (periodInDays - 2) * 1.50m;
            return result;
        }

        public int GetFrequentRenterPoints(int daysRented)
        {
            return 1;
        }

    }
}

File: Agile Development Practices/Code/Java/videoworld/src/main/java/com/thoughtworks/videorental/domain/RegularPrice.java
package com.thoughtworks.videorental.domain;

public class RegularPrice implements Price {

	public double getCharge(final int daysRented) {
		double result = 2;
		if (daysRented > 2)
			result += (daysRented - 2) * 1.5;
		return result;
	}

	public int getFrequentRenterPoints(final int daysRented) {
		return 1;
	}

}
